lantiq/xrx200: add support for the Fritzbox 7360v2

Rising-Sun opened the pull request:
This commit adds support for the Fritzbox 7360v2.

CPU: VR9 500MHz Cores: 2
RAM: 128 MB
NOR-Flash: 32 MB
WLAN: AR9287-BL1A

DECT is not working.

Thanks Sebastian Ortwein for adding 7360SL.
The dts file is derived from 7360SL.dts and 7362SL.dts.

Firmware can be flashed with this method:

1.) Set your client IP to 192.168.178.2
2.) Power on your your Fritzbox and connect to 192.168.178.1
     via ftp in the first 5 seconds.
3.) login with adam2/adam2
4.) type into the ftp prompt:

passive
binary
debug 1
quote MEDIA FLSH // (not FLASH)
put openwrt-lantiq-xrx200-avm_fritz7360v2-squashfs-sysupgrade.bin mtd1
// using the correct location for the squashfs-sysupgrade-firmware.bin

5.) wait till red light flashing turns off.
6.) type: exit

Run tested with kernel 4.19 and 5.4 on Fritzbox 7360 V2.

Issue:
Ethernet speed is slow, (iperf between a Xiaomi mir3g
and this router results in <80Mbits throughput
with a wired cable when using the gbit ports.)

Signed-off-by: Yushi Nishida kyro2man@gmx.net

hauke commented:
This looks good to me, but please squash all the commits together into one commit.

Rising-Sun commented:
I have squashed the changes.

mwarning commented on target/linux/lantiq/image/Makefile:
```diff
@@ -665,6 +665,16 @@ define Device/avm_fritz7360sl
+  $(Device/AVM)
+  IMAGE_SIZE := 32128k
+  DEVICE_DTS := FRITZ7360V2
+  DEVICE_TITLE := AVM FRITZ!Box 7360 V2 
```
Please split up DEVICE_TITLE:
```
DEVICE_VENDOR := AVM
DEVICE_MODEL := FRITZ!Box 7360
DEVICE_VARIANT := v2
```
Thanks!

Rising-Sun commented:
done

blocktrron commented on target/linux/lantiq/base-files/etc/board.d/02_network:
```diff
@@ -158,6 +158,13 @@ avm,fritz7360sl)
 		"0:lan:3" "1:lan:4" "2:lan:2" "4:lan:1" "6t@eth0"
 	;;
 
+avm,fritz7360v2)
```
This case can be merged with the `avm,fritz7360sl` case

blocktrron commented on target/linux/lantiq/files/arch/mips/boot/dts/FRITZ7360V2.dts:
```diff
@@ -0,0 +1,90 @@
+&pcie0 {
+	gpio-reset = <&gpio 38 GPIO_ACTIVE_LOW>;
+
+	pcie@0 {
```
This subnode can be dropped entirely as all information is contained in the inherited dtsi.

blocktrron reviewed (commented):
Thanks for your PR :+1: 

I've made some suggestions below.
In addition, please have a look at your commit message and rework some formal issues it currently contains. The Submitting patches guide could be helpful for this task: https://openwrt.org/submitting-patches

Rising-Sun commented:
> In addition, please have a look at your commit message and rework some formal issues it currently contains. The Submitting patches guide could be helpful for this task: https://openwrt.org/submitting-patches

Hope I have corrected everything now.

Rising-Sun commented:
I don't know how to resolve this new conflict corrcetly/easily. Can I close and open a new ticket after I do a new fork?

hauke commented:
@Rising-Sun You can do a "git rebase origin/master" and fix the merge problem and then do a force push

adrianschmutzler commented:
@Rising-Sun Please update/rebase your patches as requested.
Git rebase is a little tricky when you do it for the first time, but please invest the time to learn it instead of opening a new PR, it's enormously helpful if you've unterstood it.

adrianschmutzler commented:
Please also clean up your commit title/message:
- Remove dot at the end of title
- Lower-case "add" after colon
- Tidy-up Signed-off-by
- Improve formatting of message

andifri commented:
Is there any update on that?

adrianschmutzler commented:
This is orphaned, so I close it now. Feel free to ask for reopening when there is any progress.

Rising-Sun committed:
lantiq/xrx200: add support for the Fritzbox 7360v2

This commit adds support for the Fritzbox 7360v2

CPU: VR9 500MHz Cores: 2
RAM: 128 MB
NOR-Flash: 32 MB
WLAN: AR9287-BL1A

DECT is not working.

Thanks Sebastian Ortwein for adding 7360SL.
The dts file is derived from avm_fritz7360sl.dts.

Firmware can be flashed with this method:

1.) Set your client IP to 192.168.178.2
2.) Power on your your Fritzbox and connect to 192.168.178.1
    via ftp in the first 5 seconds.
3.) login with adam2/adam2
4.) type into the ftp prompt:

passive
binary
debug 1
quote MEDIA FLSH // (not FLASH)
put openwrt-lantiq-xrx200-avm_fritz7360v2-squashfs-sysupgrade.bin mtd1
// using the correct location for the squashfs-sysupgrade-firmware.bin

5.) wait till red light flashing turns off.
6.) type: exit

Run tested with kernel 4.19 and 5.4 on Fritzbox 7360 V2.

Issue:
Ethernet speed is slow, (iperf between a Xiaomi mir3g
and this router results in <80Mbits throughput
with a wired cable when using the gbit ports.)

Signed-off-by: Yushi Nishida kyro2man@gmx.net

Rising-Sun commented:
I reseted and pushed my branch to 1756b79e233ae92bbd013280f695821731fc611e

adrianschmutzler commented:
Here, it tells me 96eb23c034916e96bb71b35f0065c18f14351cee:
https://github.com/Rising-Sun/openwrt/commits/fritzbox_7360v2

You really have to
```
git checkout 1756b79
git branch -D fritzbox_7360v2
git checkout -b fritzbox_7360v2
git push -f origin fritzbox_7360v2
*** Notify me, then after reopening: ***
git checkout 96eb23c0 (or whatever your most recent version was)
git branch -D fritzbox_7360v2
git checkout -b fritzbox_7360v2
git push -f origin fritzbox_7360v2
```

Sorry, it's not me but GitHub which makes the rules.

Rising-Sun commented:
Thanks for the how to, have done it.

adrianschmutzler commented:
Okay, now it's open again. Your second move ...

Rising-Sun commented:
ok done

Rising-Sun commented:
I jsut realised the device is outputting an error at boot time:
```
[   14.805896] irq 144: nobody cared (try booting with the "irqpoll" option)
[   14.811239] CPU: 0 PID: 664 Comm: kmodloader Not tainted 5.4.28 #0
[   14.817404] Stack : 86877d68 8007b608 806f0000 806f8ef8 80760000 806f8ec0 806f8014 87c0be1c
[   14.825756]         808a0000 87e04b14 80740ac3 80682e68 00000000 00000001 87c0bdc0 13e56dfd
[   14.834119]         00000000 00000000 808d0000 000000db 00000030 00000000 74656420 352e342e
[   14.842468]         000000db 3e800000 00000000 000c60e7 80000000 80760000 00000000 8073790c
[   14.850823]         808d0000 ffffffbf 00000001 00000000 00000001 00000010 00000000 808a0000
[   14.859179]         ...
[   14.861619] Call Trace:
[   14.864069] [<8000f5f0>] show_stack+0x30/0x100
[   14.868527] [<80588d08>] dump_stack+0x9c/0xd4
[   14.872876] [<8008010c>] __report_bad_irq+0x5c/0xf0
[   14.877738] [<8008055c>] note_interrupt+0x2c8/0x318
[   14.882612] [<8007cf94>] handle_irq_event_percpu+0x60/0x78
[   14.888095] [<8007cff4>] handle_irq_event+0x48/0x84
[   14.892968] [<80081070>] handle_level_irq+0x104/0x1b0
[   14.898017] [<8007c4d0>] generic_handle_irq+0x40/0x58
[   14.903065] [<800037a0>] ltq_hw_irq_handler+0x90/0xec
[   14.908114] [<8007c4d0>] generic_handle_irq+0x40/0x58
[   14.913173] [<805a8fd4>] do_IRQ+0x1c/0x2c
[   14.917180] [<802f59a4>] plat_irq_dispatch+0x64/0x104
[   14.922219] [<8000ab88>] except_vec_vi_end+0xb8/0xc4
[   14.927176] [<805a89d0>] _raw_spin_unlock_irqrestore+0x18/0x20
[   14.933013] [<8007f5c8>] __setup_irq+0x4d4/0x7d4
[   14.937639] handlers:
[   14.939976] [<a33ec73b>] ath_isr [ath9k]
[   14.943787] Disabling IRQ #144
```

adrianschmutzler commented:
Hi, your commit title should contain a prefix for the target "lantiq:" or "lantiq/xrx200:" and then be followed by a statement in infinite/imperative form. A working solution would e.g. be what you have as PR title now.

Rising-Sun commented:
Hey, I have changed it.

Rising-Sun commented:
Should I manually re-request a review at the rightside of this page? I can send one to mwarning and blocktrron

lynxis commented:
ping @hauke 

@Rising-Sun the performance test should work through the router much more realistic.
For example, do a switching test between the ports and do a forwarding/nat test through it.
The switching test would be 2x PC connected via cable. The performance should be near 1Gbit. 
The forwarding/nat setup one port as dhcp port, setup a dhcp server, do an iperf between 2x PC while the fritzbox is routing+nat.

lynxis commented on target/linux/lantiq/files-5.4/arch/mips/boot/dts/lantiq/vr9_avm_fritz7360v2.dts:
```diff
@@ -0,0 +1,81 @@
+
+&state_default {
+	pcie-rst {
+		lantiq,pins = "io38";
```
Can you try using io21 here? It might fix the irq problem.

Rising-Sun replied:
I have tried io21 and I see one difference in dmesg.

io38:
`[    1.066576] pci 0000:01:00.0: [168c:ff1c] type 00 class 0x020000`
io21:
`[    1.068691] pci 0000:01:00.0: [168c:002e] type 00 class 0x028000`

So io21 is the right one?

lynxis replied:
Interesting, I thought only the message `irq 144: nobody cared (try booting with the "irqpoll" option)` would change. In your earlier dmesg the wifi card was detected as 168c:002e. Do you rebooted multiple times to get 168c:ff1c? Or was it a boot from power off.

How do you test it? Do you only booted once or rebooted multiple times?

So far, atm I would like to use io21.

Rising-Sun replied:
First boot after flashing I get 168c:ff1c with io38 and 168c:002e with io21.
After a reboot io38 boots directly with 168c:002e.

lynxis replied:
Thanks for looking into it even more! It looks like, the pcie is not a good indicator for it. I would guess the different pci ids comes from the calibration data, which got loaded when OpenWrt boots.

Rising-Sun commented:
changed io38 to io21

lynxis commented:
Thanks a lot. I've merged it, because I think it's good enough. The fritzbox is using the owl-loader which also loads the pci id. So the reset lines doesn't seems to have anything todo with it.
If you like to investigate further if io21 or io38 is correct, the pci reset should reset all wifi operations. So pulling the reset line manually (via a gpio tool or by adding the a devicetree section which pulls the ioline to low should disable the wifi.

adrianschmutzler commented on target/linux/lantiq/image/vr9.mk:
```diff
@@ -127,6 +127,16 @@ define Device/avm_fritz7360sl
+  DEVICE_VARIANT := v2
+  IMAGE_SIZE := 32128k
+  DEVICE_PACKAGES := kmod-ath9k kmod-owl-loader wpad-basic kmod-usb-dwc2
+  SUPPORTED_DEVICES += FRITZ7360V2
```
Is that required for some reason? I do not see a reason, so I'd remove it with a separate patch.

Despite, I'd like to introduce a hyphen before the v2 in the device node name/compatible ...
